cli/command/image/build.go

```diff
@@ -615,6 +625,6 @@ func imageBuildOptions(dockerCli command.Cli, options buildOptions) types.ImageB
 		Squash:         options.squash,
 		ExtraHosts:     options.extraHosts.GetAll(),
 		Target:         options.target,
-		Platform:       options.platform,
+		Platform:       &platform,
```

Comment:
Right, so need to change this, and don't set `Platform` if `options.platform` is empty